src/EFCore.PG/Infrastructure/Internal/NpgsqlOptionsExtension.cs

```diff
@@ -118,6 +127,19 @@ public override bool ApplyServices(IServiceCollection services)
         // See https://github.com/aspnet/EntityFrameworkCore/pull/10091
         public override int? MinBatchSize => base.MinBatchSize ?? 2;
 
+        /// <summary>
+        /// Returns a copy of the current instance configured with the specified range mapping.
+        /// </summary>
+        [NotNull]
+        internal virtual NpgsqlOptionsExtension WithRangeMapping(Type elementClrType, string rangeName, string subtypeName)
+        {
+            var clone = (NpgsqlOptionsExtension)Clone();
+
+            clone.RangeMappings.Add((elementClrType, rangeName, subtypeName));
```

Comment:
Shouldn't this point to the backing field?

Reply:
Thanks, definitely - and in fact the publicly-accessible `RangeMappings` was a List instead of an IReadOnlyList.